### PR TITLE
Ensure profileId is set when testing API token

### PR DESCRIPTION
**Pull request by Owaan**
Resolves [#73](https://github.com/the-hideout/TarkovMonitor/issues/73)

Not sure if the profile ID needs to be mapped to the API key as TarkovTracker isn't separating profiles (yet). This should do the job though 👍 

This also fixes the scenario where a user could be confused launching TarkovMonitor before launching Tarkov or having a profile reported in their logs. You'll still need to handle launching TarkovMonitor without even having Tarkov installed though....

### CODE REVIEW

**Comment by Owaan** [comment]
@Razzmatazzz All good!

I've fixed two issues in this PR caused by the `profileId` changes. My general thoughts are that both of these issues are caused by using a dictionary to map a `profileId` to the API tokens when you might not need to. I have left that the same in my changes incase you had future plans regarding that.

### Issue 1:
Fixed by [this commit](https://github.com/the-hideout/TarkovMonitor/pull/77/commits/446481e170ef814f80b5412c491545e49295b6f7)

**Testing Steps:**
1. Remove `AppData/Local/TarkovMonitor/<path>/<version>/user.config`
2. Launch TarkovMonitor
3. Paste TarkovTracker API key into Settings API Key 
4. Click "Test Token"

**Expected Result:**
API success response

**Actual Result:**
"TarkovTracker API error: The given key '' was not present in the dictionary."

**Solution:**
The code here, `currentProfile` could be an empty string so you'd never find the token in the dictionary.
```C#
return Task.Run<string>(() => {
    return tokens[currentProfile];
});
```
I'm now passing the profileId into the `TestToken` method.

### Issue 2:
Fixed by [this commit](https://github.com/the-hideout/TarkovMonitor/pull/77/commits/7c4390d40b2f3aec508400f668f70a7135015b15)

**Testing Steps:**
1. Remove `AppData/Local/TarkovMonitor/<path>/<version>/user.config`
2. Empty your `C:\Battlestate Games\EFT\Logs` directory (leaving one log which doesn't include a profileId)
3. Launch TarkovMonitor

**Expected Result:**
A graceful error indicating the game needs to be started at least once to track progress

**Actual Result:**
An ambiguous error (I can't even recall exactly what it was at this point)

**Solution:**
I've just tried to catch a few places to handle if the `profileId` couldn't be found.

**Comment by Razzmatazzz** [comment]
Thanks for the explanation, and for the fixes!

File: TarkovMonitor/TarkovTracker.cs [hunk 1]
```diff
@@ -15,7 +15,7 @@ internal interface ITarkovTrackerAPI
 
             [Get("/token")]
             [Headers("Authorization: Bearer {token}")]
-            Task<TokenResponse> TestToken(string token);
```
Comment (Razzmatazzz) [inline TarkovMonitor/TarkovTracker.cs:18]: It isn't necessary to pass the profile id here.

File: TarkovMonitor/TarkovTracker.cs [hunk 2]
```diff
@@ -94,7 +102,7 @@ public static async Task<ProgressResponse> SetProfile(string profileId)
                 Progress = new();
                 return Progress;
             }
-            await TestToken(newToken);
+            await TestToken(profileId, newToken);
```
Comment (Razzmatazzz) [inline TarkovMonitor/TarkovTracker.cs:105]: profileId not needed here

File: TarkovMonitor/TarkovTracker.cs [hunk 3]
```diff
@@ -285,11 +293,11 @@ public static async Task<ProgressResponse> GetProgress()
             }
         }
 
-        public static async Task<TokenResponse> TestToken(string apiToken)
+        public static async Task<TokenResponse> TestToken(string profileId, string apiToken)
```
Comment (Razzmatazzz) [inline TarkovMonitor/TarkovTracker.cs:296]: profileId also not needed here

**Review by Razzmatazzz — CHANGES_REQUESTED** [review]
Unless I'm misunderstanding things, I think some of these changes can be streamlined a bit. Thanks again for the PR!

**Comment by Owaan** [comment]
@Razzmatazzz oh yeah you might be right I think I misread something and also forgot to use the profileId I passed in anyways lol... Will clean it up in a bit, test and recommit

**Comment by Owaan** [comment]
@Razzmatazzz All changes fixed, first time using C#, apologies that wasn't so clean the first go around

I've just used your existing `GetToken` method in the `AuthorizationHeaderValueGetter` method now.

The main issue was in `InitializeProgress`, we needed to set the `TarkovTracker` profile before calling `TarkovTracker.TestToken()`

**Review by Razzmatazzz — APPROVED** [review]
Looks great, thanks again!

**Pull request merged**